spec/statement.dd

```diff
@@ -835,52 +835,53 @@ void main()
         that is confusing to read. Therefore, using $(D foreach_reverse) with a
         delegate is now deprecated, and will be rejected in the future.)
 
-$(H3 $(LNAME2 foreach_over_tuples, Foreach over Tuples))
+$(H3 $(LNAME2 foreach_over_tuples, Foreach over Sequences))
 
 $(P
-        If the aggregate expression is a tuple, there
-        can be one or two variables declared. If one, then the variable
-        is said to be the $(I value) set to the elements of the tuple,
-        one by one. If the type of the
-        variable is given, it must match the type of the tuple contents.
-        If it is not given, the type of the variable is set to the type
-        of the tuple element, which may change from iteration to iteration.
-        If there are
-        two variables declared, the first is said to be the $(I index)
-        and the second is said to be the $(I value). The $(I index)
-        must be of $(D int) or $(D uint) type, it cannot be $(I ref),
-        and it is set to be the index of the tuple element.
-)
-
-$(P
-        If the tuple is a list of types, then the foreach statement
-        is executed once for each type, and the value is aliased to that
+        If the aggregate expression is a sequence, there
+        can be one or two iteration symbols declared. If one, then the symbol
+        is an $(I element alias) of each element in the sequence in turn.
+)$(P
+        If the sequence is a $(I TypeSeq), then the foreach statement
+        is executed once for each type, and the element alias is set to each
         type.
+)$(P
+        When the sequence is a $(I ValueSeq), the element alias is a variable
+        and is set to each value in the sequence. If the type of the
+        variable is given, it must match the type of the sequence contents.
+        If no type is given, the type of the variable is set to the type
+        of the sequence element, which may change from iteration to iteration.
+)$(P
+        If there are
+        two symbols declared, the first is the $(I index variable)
+        and the second is the $(I element alias). The index
+        must be of $(D int) or $(D uint) type, it cannot be `ref`,
```

Comment:
It can also be `long`: https://run.dlang.io/is/kxSAFP as the index is `size_t`

Reply:
@wilzbach I've added `long` and `ulong` here, and updated runtime foreach index types, which listed just `size_t` instead of these. BTW for runtime foreach only, dmd accepts `short` and `byte` - I've not documented this though.